libs/ssh/server.go

```diff
@@ -0,0 +1,322 @@
+package ssh
+
+import (
+	"context"
+	_ "embed"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"os/user"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/databricks/cli/libs/cmdio"
+	"github.com/databricks/databricks-sdk-go"
+	"golang.org/x/sync/errgroup"
+)
+
+//go:embed jupyter-init.py
+var jupyterInitScript string
+
+// connectionsManager manages concurrent websocket clients and sends a shutdown signal if no
+// clients are connected for a specified duration.
+type connectionsManager struct {
+	maxClients      int
+	shutdownDelay   time.Duration
+	shutdownTimer   *time.Timer
+	shutdownTimerMu sync.Mutex
+	connections     map[string]*proxyConnection
+	connectionsMu   sync.Mutex
+	TimedOut        chan bool
+}
+
+func newConnectionsManager(maxClients int, shutdownDelay time.Duration) *connectionsManager {
+	cm := &connectionsManager{
+		maxClients:    maxClients,
+		shutdownDelay: shutdownDelay,
+		connections:   make(map[string]*proxyConnection),
+		TimedOut:      make(chan bool),
+	}
+	cm.startShutdownTimer()
+	return cm
+}
+
+func (cm *connectionsManager) TryAdd(id string, conn *proxyConnection) bool {
+	if len(cm.connections) >= cm.maxClients {
+		return false
+	}
+	cm.addConnection(id, conn)
+	cmdio.LogString(context.Background(), fmt.Sprintf("Client connected, current count: %d", len(cm.connections)))
```

Comment:
I recommend using the logger for this type of stuff.

Then you can always run the server with `--log-file /Workspace/.../.ssh/debug.log` to write debug logs.